packages/url_launcher/url_launcher_ios/test/url_launcher_ios_test.dart

```diff
@@ -134,11 +135,171 @@ void main() {
+      final bool launched = await launcher.launchUrl('http://example.com/',
+          const LaunchOptions(mode: PreferredLaunchMode.inAppWebView));
+      expect(launched, true);
+      expect(api.usedSafariViewController, true);
+    });
+
+    test('calls through with inAppBrowserView', () async {
+      final UrlLauncherIOS launcher = UrlLauncherIOS(api: api);
+      final bool launched = await launcher.launchUrl('http://example.com/',
+          const LaunchOptions(mode: PreferredLaunchMode.inAppBrowserView));
+      expect(launched, true);
+      expect(api.usedSafariViewController, true);
+    });
+
+    test('passes invalid URL PlatformException through', () async {
+      final UrlLauncherIOS launcher = UrlLauncherIOS(api: api);
+      await expectLater(
+          launcher.launchUrl('invalid://u r l',
+              const LaunchOptions(mode: PreferredLaunchMode.inAppWebView)),
+          throwsA(isA<PlatformException>()));
+    });
+  });
+
+  group('launch with universal links', () {
+    test('calls through', () async {
+      final UrlLauncherIOS launcher = UrlLauncherIOS(api: api);
+      final bool launched = await launcher.launchUrl(
+        'http://example.com/',
+        const LaunchOptions(
+            mode: PreferredLaunchMode.externalNonBrowserApplication),
+      );
+      expect(launched, true);
+      expect(api.usedSafariViewController, false);
+      expect(api.passedUniversalLinksOnly, true);
+    });
+
+    test('passes invalid URL PlatformException through', () async {
+      final UrlLauncherIOS launcher = UrlLauncherIOS(api: api);
+      await expectLater(
+          launcher.launchUrl(
+              'invalid://u r l',
+              const LaunchOptions(
+                  mode: PreferredLaunchMode.externalNonBrowserApplication)),
+          throwsA(isA<PlatformException>()));
+    });
+  });
+
+  group('launch with platform default', () {
+    test('uses Safari view controller for http', () async {
+      final UrlLauncherIOS launcher = UrlLauncherIOS(api: api);
+      final bool launched = await launcher.launchUrl(
+          'http://example.com/', const LaunchOptions());
+      expect(launched, true);
+      expect(api.usedSafariViewController, true);
+    });
+
+    test('uses Safari view controller for https', () async {
+      final UrlLauncherIOS launcher = UrlLauncherIOS(api: api);
+      final bool launched = await launcher.launchUrl(
+          'https://example.com/', const LaunchOptions());
+      expect(launched, true);
+      expect(api.usedSafariViewController, true);
+    });
+
+    test('uses standard external for other schemes', () async {
+      final UrlLauncherIOS launcher = UrlLauncherIOS(api: api);
+      final bool launched = await launcher.launchUrl(
+          'supportedcustomscheme://example.com/', const LaunchOptions());
+      expect(launched, true);
+      expect(api.usedSafariViewController, false);
+      expect(api.passedUniversalLinksOnly, false);
+    });
+  });
```

Comment:
As with Android tests, the above is copies of the existing tests but with the non-deprecated API.